avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -544,11 +569,24 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
   }
 
   public org.apache.avro.Schema fromConnectSchema(Schema schema) {
+    return fromConnectSchema(schema, new HashMap<String, org.apache.avro.Schema>());
+  }
+
+  public org.apache.avro.Schema fromConnectSchema(Schema schema, Map<String, org.apache.avro.Schema> schemaMap) {
+    return fromConnectSchema(schema, schemaMap, false);
+  }
+
+
+  public org.apache.avro.Schema fromConnectSchema(Schema schema, Map<String, org.apache.avro.Schema> schemaMap, boolean ignoreOptional) {
     if (schema == null) {
       return ANYTHING_SCHEMA;
     }
 
     org.apache.avro.Schema cached = fromConnectSchemaCache.get(schema);
+
+    if (cached == null && !AVRO_TYPE_UNION.equals(schema.name()) && !schema.isOptional()) {
```

Comment:
Ok, this makes sense as the map is actually specific to the transformation of an individual schema. Maybe just add a comment somewhere explaining what it is used for?